### PR TITLE
Allow generics <TValue, TError> on ValueOrError type

### PR DESCRIPTION
**Pull request by Odonno**
Linked to #48 

It was relatively easy to make the change but I got mad with all these generics everywhere. :) Maybe there is a better way, for now I followed the easy one.

Also, I have 16 failed tests but not directly related to `ValueOrError` type. Not sure if it is my machine or if the tests are really broken.

### CODE REVIEW

File: src/SuccincT/Options/ValueOrError(TV).cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace SuccincT.Options
+{
+    public class ValueOrError<TValue> : ValueOrError<TValue, string>
```
Comment (Odonno) [inline src/SuccincT/Options/ValueOrError(TV).cs]: I think it is useless. Either use full types or default type (string, string).

**Comment by Odonno** [comment]
Ok. Seems like 3 unit tests are broken due to JSON conversion. I will change that.

**Comment by Odonno** [comment]
@DavidArno Well, I made the change on JSON serialization library but I needed to use reflexion methods only available on .NET Standard 2.0. So, it's a breaking change.

And I also need to add more unit tests.

**Comment by Odonno** [comment]
Still pending.

**Comment by DavidArno** [comment]
It is indeed. I've been thrashing around with v4 of Succinct for months now. I have multiple branches, with multiple experiments with read-only structs and nullable reference types enabled, all in search of something I'm happy with.


With C# 8 due in the next month, I've created yet another branch and have started the whole process again. So v4 is planned for late September/early October when C# 8 goes live.

**Comment by Odonno** [comment]
@DavidArno Good to know! Thank you for your dedicated work.

**Pull request closed without merging**